lib/mix/lib/mix/tasks/test.ex

```diff
@@ -277,6 +279,9 @@ defmodule Mix.Tasks.Test do
         option_only_present? = Keyword.has_key?(opts, :only)
 
         cond do
+          failures >= opts[:max_fail] ->
```

Comment:
This is only invoked after all tests run, so it is not going to give you the desired effect.

Reply:
Right, it will be caught and it won't stop until the suite is done. Should a call to `agent_stop` be made?

Reply:
@Kartstig we would need to add support for this in ExUnit instead.

Reply:
If max fail is zero, this will fail eventhough no failures happened. So we need to cover this corner case

Reply:
Actually the logic is wrong. It should be `>`